### PR TITLE
Chore: #17 Update project setup to pyproject.toml

### PR DESCRIPTION
**Pull request by fedorgrab**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ $ pip install git+https://github.com/broadinstitute/cell-annotation-service-clie
 To use Cellarium CAS, create a client instance with your API token:
 
 ```python3
-from cas_cli import CASClient
+from cas import CASClient
```
Comment (ordabayevy) [inline README.md:12]: -> `from cellarium.cas import CASClient`

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+optional-dependencies = {test = { file = ["requirements/test.txt"] }}
+readme = {file = ["README.md"]}
+
+[project.urls]
+"Homepage" = "https://cellarium.ai"
+"Source Code" = "https://github.com/cellarium-ai/cellarium-cas"
+"Bug Tracker" = "https://github.com/cellarium-ai/cellarium-cas/issues"
+"Discussion Forum" = "https://github.com/cellarium-ai/cellarium-cas/discussions"
+
+[tool.setuptools-git-versioning]
```
Comment (ordabayevy) [inline pyproject.toml:34]: Remove it or change to `setuptools-scm`?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+enabled = true
+
+[tool.ruff]
```
Comment (ordabayevy) [inline pyproject.toml:37]: Remove this as well if you are not using `ruff` linter.

File: cellarium/__init__.py
Comment (ordabayevy) [inline cellarium/__init__.py:1]: It seems like you can remove this file. See https://packaging.python.org/en/latest/guides/packaging-namespace-packages/

File: pyproject.toml (hunk 2 above)
Comment (fedorgrab) [reply pyproject.toml:34]: Good catch

**Comment by fedorgrab** [comment]
@ordabayevy Just resolved conflicts

**Review by ordabayevy — APPROVED** [review]
LGTM!

**Pull request merged**